DOCS-6279 Adds Getting Started with CI Visibility Documentation (#22692)
DataDog/documentation: content/en/getting_started/ci_visibility/_index.md

You can forward CI job logs and automatically correlate them with your pipelines in CI Visibility. Depending on the providers you are using, you can either enable job log collection on the [**Settings** page][1] in CI Visibility or in your provider’s settings to integrate with Datadog.

You can also use the `datadog-ci` CLI to [trace commands][2] in your pipelines, as well as the [custom tags and measures commands][3] to add user-defined text and numerical tags in your pipeline traces.

CI Visibility provides DevOps and platform engineering organizations with comprehensive monitoring, analytics, and the ability to pinpoint and resolve bottlenecks, optimize resource allocation, and decrease pipeline costs. 

By integrating performance metrics, logs, and alerts, your organization can maintain development speed, improve the reliability of your pipelines, and make data-informed decisions across cloud and self-hosted environments.

## Set up your CI provider

CI Visibility tracks the performance and results of your CI pipelines, and displays results after the pipeline finishes. 

> To start sending pipeline metrics, see the documentation for one of the following CI providers. 

Reviewer: Should we mention what they can do if Datadog does not support their CI provider? Manual steps, if any?

{{< partial name="getting_started/ci_visibility/providers.html" >}}

</br>

> Depending on the CI provider(s) of choice, CI Visibility may not support all of the levels in your pipeline (stage, job, step, or command). For more information about how CI Visibility defines a CI pipeline, see the [Terminology section][4].

Author: ```suggestion
For a CI provider that is not supported by Datadog, you can send your pipeline events programmatically using the [CI Visibility Pipelines API].

Depending on the CI provider(s) of choice, CI Visibility may not support all of the levels in your pipeline (stage, job, step, or command). For more information about how CI Visibility defines a CI pipeline, see the [Terminology section][4].
```

{{< img src="/getting_started/ci_visibility/avg_duration_monitor.png" alt="A CI pipeline monitor configured to trigger an alert when the average duration for the Test and Deploy Cart pipeline exceeds five minutes in the past day" style="width:100%" >}}

To set up a monitor that alerts on your CI pipeline when the average duration in the past day exceeds a five minute threshold:

1. Navigate to [**Monitors** > **New Monitor**][15] and select **CI**. 
1. Select a common monitor type for CI pipelines to get started, for example: `Long Running Pipeline` to trigger alerts when a pipeline has been running for too long or `Failed Job` to trigger alerts for job failures, or customize your own search query. In this example, enter `@ci.pipeline.name:test_and_deploy_cart` and select the Avg of `Duration (@duration)`.
1. In the `Evaluate the query over the` section, select **last 1 day**. 
1. Set the alert conditions to trigger when the evaluated value is **above** the threshold, and specify values for the alert or warning thresholds, such as `Alert threshold > 300000000000`.

> 1. Define the monitor's notification.

Reviewer: ```suggestion
1. In the `Notify your team` section, configure your monitor's notification settings.
```

1. Set permissions for the monitor.
1. Click **Create**.

## Further Reading

{{< partial name="whats-next/whats-next.html" >}}

[1]: https://app.datadoghq.com/ci/settings
[2]: /continuous_integration/pipelines/custom_commands/
[3]: /continuous_integration/pipelines/custom_tags_and_measures/
[4]: /continuous_integration/pipelines/?tab=githubactions#terminology
[5]: /dashboards/
[6]: https://app.datadoghq.com/dash/integration/30516/ci-visibility---pipelines-dashboard
[7]: https://app.datadoghq.com/ci/pipelines
[8]: /continuous_integration/search/
[9]: /continuous_integration/explorer?tab=pipelineexecutions
[10]: /tests/guides/flaky_test_management/
[11]: /continuous_integration/pipelines/?tab=githubactions#supported-features
[12]: /continuous_integration/guides/infrastructure_metrics_with_gitlab/
[13]: /continuous_integration/guides/pipeline_data_model/
[14]: /monitors/types/ci/?tab=pipelines

> [15]: https://app.datadoghq.com/monitors/create/

Reviewer: Link doesn't work when there is a slash at the end

```suggestion
[15]: https://app.datadoghq.com/monitors/create
```

You can also use the `datadog-ci` CLI to [trace commands][2] in your pipelines, as well as the [custom tags and measures commands][3] to add user-defined text and numerical tags in your pipeline traces.

CI Visibility provides DevOps and platform engineering organizations with comprehensive monitoring, analytics, and the ability to pinpoint and resolve bottlenecks, optimize resource allocation, and decrease CI costs. 

By integrating performance metrics, logs, and alerts, organizations can improve development speed, increase the reliability of their pipelines, and make data-informed decisions across cloud and self-hosted environments.

## Set up your CI provider

CI Visibility tracks the performance and results of your CI pipelines, and displays results after the pipeline finishes. 

To start sending pipeline metrics, see the documentation for one of the following CI providers that Datadog supports below.

{{< partial name="continuous_integration/ci-pipelines-getting-started.html" >}}

</br>

> If your CI provider is not supported, you can programmatically send your pipeline events to Datadog using the [CI Visibility Pipelines API][16].

Author: ```suggestion
If your CI provider is not supported, you can programmatically send your pipeline events to Datadog. See [Send pipeline events to Datadog](#send-pipeline-events-to-datadog).
```

{{< img src="/getting_started/ci_visibility/executions.png" alt="Pipeline execution results visualized as a flame graph for the Staging Build and Test pipeline" style="width:100%" >}}

You can identify bottlenecks in your pipeline and examine individual nodes ranked from the largest to smallest percentage of execution time. 

After you have set up Test Visibility, you can access information about tests that were run in your CI pipelines, including the test status (Failed, New Flaky, Passed, or Skipped), on the Test Runs tab in a pipeline execution’s side panel. For more information, see the [Flaky Test Management documentation][10].

You can access pipeline or job logs across cloud and self-hosted runners and see information about your runners on the Logs tab in a pipeline execution’s side panel.

If you are using [supported providers][11], you can correlate infrastructure metrics with your GitLab jobs and access the GitLab job’s host, system, host tags, and host metrics information. For more information, see [Correlate Infrastructure Metrics with GitLab Jobs in Datadog][12].

## Send pipeline events to Datadog

> For other pipeline providers and custom pipelines, you can programmatically send pipeline events to Datadog using the CI Visibility Pipelines API endpoint. For more information, see [Pipeline Data Model and Execution Types][13].

Author: ```suggestion
For other pipeline providers and custom pipelines, you can programmatically send pipeline events to Datadog using the [CI Visibility Pipelines API endpoint][16]. For more information, see [Pipeline Data Model and Execution Types][13].
```
